src/hotspot/cpu/x86/stubGenerator_x86_64_poly.cpp

```diff
@@ -0,0 +1,1066 @@
+  __ vpinsrq(D2, D2, a1, 1);
+  __ vinserti32x4(D0, D0, D2, 0);
+  __ movq(D2, a2);
+  __ vinserti32x4(D1, D1, D2, 0);
+
+  // Interleave the powers of R^1..R^4 to form 44-bit limbs (half-empty)
+  // B0 to have bits 0-43 of all 4 blocks in alternating 8 qwords
+  // B1 to have bits 87-44 of all 4 blocks in alternating 8 qwords
+  // B2 to have bits 127-88 of all 4 blocks in alternating 8 qwords
+  __ vpxorq(D2, D2, D2, Assembler::AVX_512bit);
+  poly1305_limbs_avx512(D0, D2, B0, B1, B2, false, TMP, rscratch);
+
+  // D1 contains the 2 highest bits of the powers of R
+  __ vpsllq(D1, D1, 40, Assembler::AVX_512bit);
+  __ evporq(B2, B2, D1, Assembler::AVX_512bit);
+
+  // Broadcast 44-bit limbs of R^4 into R0,R1,R2
+  __ mov(t0, a0);
+  __ andq(t0, ExternalAddress(poly1305_mask44()), rscratch); // First limb (R^4[43:0])
+  __ evpbroadcastq(R0, t0, Assembler::AVX_512bit);
+
+  __ movq(t0, a1);
+  __ shrdq(a0, t0, 44);
+  __ andq(a0, ExternalAddress(poly1305_mask44()), rscratch); // Second limb (R^4[87:44])
+  __ evpbroadcastq(R1, a0, Assembler::AVX_512bit);
+
+  __ shrdq(a1, a2, 24);
+  __ andq(a1, ExternalAddress(poly1305_mask42()), rscratch); // Third limb (R^4[129:88])
+  __ evpbroadcastq(R2, a1, Assembler::AVX_512bit);
+
+  // Generate 4*5*R^4 into {R2P,R1P}
+  // Used as multiplier in poly1305_multiply8_avx512 so can
+  // ignore bottom limb and carry propagation
+  __ vpsllq(R1P, R1, 2, Assembler::AVX_512bit);    // 4*R^4
+  __ vpsllq(R2P, R2, 2, Assembler::AVX_512bit);
+  __ vpaddq(R1P, R1P, R1, Assembler::AVX_512bit);  // 5*R^4
+  __ vpaddq(R2P, R2P, R2, Assembler::AVX_512bit);
+  __ vpsllq(R1P, R1P, 2, Assembler::AVX_512bit);   // 4*5*R^4
+  __ vpsllq(R2P, R2P, 2, Assembler::AVX_512bit);
+
+  // Move R^4..R^1 one element over
+  __ vpslldq(C0, B0, 8, Assembler::AVX_512bit);
+  __ vpslldq(C1, B1, 8, Assembler::AVX_512bit);
+  __ vpslldq(C2, B2, 8, Assembler::AVX_512bit);
+
+  // Calculate R^8-R^5
+  poly1305_multiply8_avx512(B0, B1, B2,             // ACC=R^4..R^1
+                            R0, R1, R2, R1P, R2P,   // R^4..R^4, 4*5*R^4
+                            T0, T1, T2, T3, T4, T5, TMP, rscratch);
+
+  // Interleave powers of R: R^8 R^4 R^7 R^3 R^6 R^2 R^5 R
+  __ evporq(B0, B0, C0, Assembler::AVX_512bit);
+  __ evporq(B1, B1, C1, Assembler::AVX_512bit);
+  __ evporq(B2, B2, C2, Assembler::AVX_512bit);
+
+  // Broadcast R^8
+  __ vpbroadcastq(R0, B0, Assembler::AVX_512bit);
+  __ vpbroadcastq(R1, B1, Assembler::AVX_512bit);
+  __ vpbroadcastq(R2, B2, Assembler::AVX_512bit);
+
+  // Generate 4*5*R^8
+  __ vpsllq(R1P, R1, 2, Assembler::AVX_512bit);
+  __ vpsllq(R2P, R2, 2, Assembler::AVX_512bit);
+  __ vpaddq(R1P, R1P, R1, Assembler::AVX_512bit);    // 5*R^8
+  __ vpaddq(R2P, R2P, R2, Assembler::AVX_512bit);
+  __ vpsllq(R1P, R1P, 2, Assembler::AVX_512bit);     // 4*5*R^8
+  __ vpsllq(R2P, R2P, 2, Assembler::AVX_512bit);
+
+  // Store R^8-R for later use
+  __ evmovdquq(Address(rsp, 64*0), B0, Assembler::AVX_512bit);
```

Comment:
Could these vector spills be eliminated? I counted 8 spare zmm registers available across the vector loop (xmm7-xmm12, xmm30, xmm31).

And here's what is explicitly used in `process256Loop`:
```
D0 D1              = xmm2-xmm3
B0 B1 B2 B3 B4 B5  = xmm19-xmm24
TMP                = xmm6
A0 A1 A2 A3 A4 A5  = xmm13-xmm18
R0 R1 R2 R1P R2P   = xmm25-xmm29
T0 T1 T2 T3 T4 T5  = xmm0-xmm5
```

Reply:
Interesting!! Let me try that!

Reply:
Done!

PS: This find really was great!
PPS: I also reordered the map alphabetically and counted in-order... it was just really bugging me!!